### ISSUE TITLE
Gearman::Worker documentation

### ISSUE THREAD

**Issue by esabol** [p-alik/perl-Gearman#41]
Found a typo. Please change "reset_abilityes" to "reset_abilities" here:

https://github.com/p-alik/perl-Gearman/blob/cc1f076920938a69ccafb3dbf9d7202c37508836/lib/Gearman/Worker.pm#L182

Also, I feel that the `stop_if`, `on_start`, `on_complete` and `on_fail` callbacks for `work` need to be documented better. `stop_if` has an example, but the others are only barely mentioned. It's not clear what values are being passed to these callbacks unless you look at the source code (and even then it's not very clear).

A useful example might be a worker that stops working after N completed (or failed?) jobs.

**Comment by esabol** [comment]
Looking better!

i also recommend changing:

https://github.com/p-alik/perl-Gearman/blob/1652870e96f665302682785f211c7979eeb426c2/lib/Gearman/Worker.pm#L195

to

```
This tells all the job servers that this worker can no longer do any tasks.
```
And
https://github.com/p-alik/perl-Gearman/blob/1652870e96f665302682785f211c7979eeb426c2/lib/Gearman/Worker.pm#L213

to

```
This endlessly loops. It takes an applicable job, if available, does the job, and then waits for the next one.
```

And
https://github.com/p-alik/perl-Gearman/blob/1652870e96f665302682785f211c7979eeb426c2/lib/Gearman/Worker.pm#L130-L131

to

```
If you intend to send or receive UTF-8 data over SSL connections,
beware that there is no UTF-8 support in the underlying L<Net::SSLeay>.
```

**Comment by p-alik** [comment]
I really appreciate your suggestions @esabol. Thank you very much!

**Comment by p-alik** [comment]
enhancement is released in 2.004.015
https://metacpan.org/changes/release/PALIK/Gearman-2.004.015

thank you
